services/workers/src/test/java/org/sagebionetworks/object/snapshot/worker/utils/NodeObjectRecordBuilderTest.java

```diff
@@ -47,10 +83,22 @@ public void invalidObjectType() {
 
 	@Test
 	public void validChangeMessage() {
+		// Restricted
+		List<AccessRequirement> ars = new ArrayList<AccessRequirement>();
+		ars.add(new PostMessageContentAccessRequirement());
+		Mockito.when(mockArs.getResults()).thenReturn(ars);
+		// not public
+		Mockito.when(mockPermissions.getCanPublicRead()).thenReturn(false);
+
 		Long timestamp = System.currentTimeMillis();
 		Message message = MessageUtils.buildMessage(ChangeType.UPDATE, "123", ObjectType.ENTITY, "etag", timestamp);
 		ChangeMessage changeMessage = MessageUtils.extractMessageBody(message);
+
+		node.setIsPublic(false);
+		node.setIsControlled(false);
+		node.setIsRestricted(true);
 		ObjectRecord expected = ObjectRecordBuilderUtils.buildObjectRecord(node, timestamp);
+
 		ObjectRecord actual = builder.build(changeMessage);
 		Mockito.verify(mockNodeDAO).getNode(Mockito.eq("123"));
 		assertEquals(expected, actual);
```

Comment:
There were three new switches added but I only see a test for one of the three.  Expected a test for public true & false, controlled true and false, and restricted true and false.